zapcore/json_encoder_impl_test.go

```diff
@@ -191,6 +191,18 @@ func TestJSONEncoderObjectFields(t *testing.T) {
 				assert.Error(t, e.AddReflected("k", noJSON{}), "Unexpected success JSON-serializing a noJSON.")
 			},
 		},
+		{
+			desc: "namespace",
```

Comment:
probably useful in covering more cases, such as:
- `ns1`
- `ns1, val`
- `ns1, ns2, val`
- `ns1, val, ns2, val2 // this is the one you have now`

I guess if you want fewer cases, let's have: `ns1, ns2, val, ns3, val2, ns4` since it captures:
- a ns with none of its own fields
- an empty leaf ns
- still capture values at multiple levels